app/main.py

```diff
@@ -1,3 +1,72 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        raise TypeError("Operand must be of type Distance, int, or float")
+
+    def __iadd__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Operand must be of type Distance, int, or float")
+        return self
+
+    def __mul__(self, other: "Distance") -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        raise TypeError("Operand must be of type int or float")
+
+    def __truediv__(self, other: "Distance") -> "Distance":
```

Comment:
The `__truediv__` method should have the parameter `other` typed as `Union[int, float]` instead of `Distance`, since it only supports division by numbers.